app/src/main/java/com/google/samples/apps/sunflower/compose/garden/GardenScreen.kt

```diff
@@ -0,0 +1,263 @@
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Card
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.pluralStringResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import com.google.samples.apps.sunflower.viewmodels.GardenPlantingListViewModel
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.bumptech.glide.integration.compose.ExperimentalGlideComposeApi
+import com.bumptech.glide.integration.compose.GlideImage
+import com.google.accompanist.themeadapter.material.MdcTheme
+import com.google.samples.apps.sunflower.R
+import com.google.samples.apps.sunflower.compose.card
+import com.google.samples.apps.sunflower.data.GardenPlanting
+import com.google.samples.apps.sunflower.data.Plant
+import com.google.samples.apps.sunflower.data.PlantAndGardenPlantings
+import com.google.samples.apps.sunflower.viewmodels.PlantAndGardenPlantingsViewModel
+import java.util.*
+
+@Composable
+fun GardenScreen(
+    viewModel: GardenPlantingListViewModel = viewModel(),
+    onAddPlantClick: () -> Unit,
+    onPlantClick: (PlantAndGardenPlantings) -> Unit
+) {
+    val gardenPlants = viewModel.plantAndGardenPlantings.observeAsState().value
+    GardenScreen(
+        gardenPlants = gardenPlants,
+        onAddPlantClick = onAddPlantClick,
+        onPlantClick = onPlantClick
+    )
+}
+
+@Composable
+private fun GardenScreen(
+    gardenPlants: List<PlantAndGardenPlantings>?,
+    onAddPlantClick: () -> Unit = {},
+    onPlantClick: (PlantAndGardenPlantings) -> Unit = {}
+) {
+    if (gardenPlants == null || gardenPlants.isEmpty()) {
+        EmptyGarden(onAddPlantClick)
+    } else {
+        GardenList(gardenPlants, onPlantClick = onPlantClick)
+    }
+
+    // When gardenPlants is not null, the content should be fully drawn
+    ReportDrawnWhen {
+        gardenPlants != null
+    }
+}
+
+@Composable
+private fun GardenList(
+    gardenPlants: List<PlantAndGardenPlantings>,
+    onPlantClick: (PlantAndGardenPlantings) -> Unit
+) {
+    LazyVerticalGrid(
```

Comment:
alternative is to call ReportDrawnWhen the grid has rendered some items
``` kotlin
val gridState = rememberLazyGridState()
ReportDrawnWhen { gridState.layoutInfo.totalItemsCount > 0 }
```

This way you'd report it more closer to when the UI is actually ready

Reply:
I like that. Going to move the `ReportDrawnWhen` calls withing `EmptyGarden` and `GardenList`